[Spring 체스 - 1단계] 쿤(홍동건) 미션 제출합니다.

Hongdonggeon opened the pull request:
안녕하세요 에단~!!
이번에 리뷰를 받게 된 쿤입니다!! 너무 영광입니다😃

Spark Java 프레임워크에서 스프링 프레임워크로 변경하기 위해 Dao, Service, Controller 를 변경해주었습니다!!

### DAO
- JDBC Template을 이용하여 Connection을 직접 만들어 주는 로직을 대체하게 했습니다.
- 빈으로 등록되게 하기 위해, @Repository 어노테이션을 붙여주었습니다.

### Service
- 빈으로 등록되게 하기 위해, @Service 어노테이션을 붙여주었습니다.
- Spring, Spark,Console 에서 공통적으로 Service를 사용하기 위해서, 외부에서 구현체를 넣어주기 위해, 생성자 구조를 변경시켜주었습니다.

### Controller
- 빈으로 등록되게 하기 위해, @Controller 어노테이션을 붙여주었습니다.
- @RequestBody와 @ResponseBody 를 사용하여 객체를 주고받게 하였습니다.
- ResponseEntity를 사용해 에러 상황에도, 에러메세지를 보이게 해주었습니다.

이전에 SparkJava를 이용하다 Spring으로 변경해보니, 너무 간편하다는게 느껴지네요!! 
리뷰 기다리면서, Spring 동작 구조 공부하고 있겠습니다!!
잘 부탁드립니다 에단 😃  마구 마구 때려주세요 🔥

Commit:
feat(init)

her0807 committed:
feat: Gson 의존성 추가

her0807 committed:
docs: 요구사항 도출

Hongdonggeon committed:
.gitignore 수정

Hongdonggeon committed:
feat: TurnDao jdbcTemplate으로 변경

Hongdonggeon committed:
feat: gameStatusDao jdbcTemplate으로 변경

Hongdonggeon committed:
move: Dto 패키지 위치 변경

Hongdonggeon committed:
feat: BoardDao jdbcTemplate으로 적용 및 RowMapper 구현

Hongdonggeon committed:
docs: 기능 요구사항 진행상황 체크

Hongdonggeon committed:
feat: JdbcTemplateTurnDao, JdbcTemplateGameStatusDao 초기화하는 기능 추가

Hongdonggeon committed:
feat: Dao 인터페이스에 초기화하는 init() 선언

Hongdonggeon committed:
feat: Dao와 Service 빈 등록을 위한 어노테이션 추가

Service 클래스 추상화를 위한 생성자 구조 변경

Hongdonggeon committed:
feat: Controller 시작화면 맵핑

데이터베이스 h2 사용

Hongdonggeon committed:
feat: Controller 전체 기능 맵핑

Hongdonggeon committed:
fix: Pawn이 대각선으로 갈 수 없는 경우에 이동하는 버그 수정

Hongdonggeon committed:
refactor: Spark Application 로직 이동

Hongdonggeon committed:
fix: 점수 출력 로직 오류 수정

Hongdonggeon committed:
docs: README 수정

her0807 committed:
feat: level1 코드 가져오기

her0807 committed:
fix: 누락된 이미지 파일 추가

her0807 committed:
fix: 누락된 도커 설정 파일 추가

her0807 committed:
fix: init.sql 추가 업로드

her0807 committed:
refactor: code style 세팅 + Posix new line 설정

Laterality commented on src/main/java/chess/dao/jdbctemplate/JdbcTemplateBoardDao.java:
```diff
@@ -0,0 +1,63 @@
+    public void init(Map<String, String> board) {
+        jdbcTemplate.execute("DROP TABLE board IF EXISTS");
+
+        jdbcTemplate.execute("create table board("
+                + " position varchar(10) not null,"
+                + " piece varchar(20) not null,"
+                + " primary key (position)"
+                + ")");
```
테이블을 삭제하고 다시 만들기 위해 이 클래스가 테이블의 상세 스키마까지 알고 있어야 할까요?

Hongdonggeon replied:
schema.sql 을 이용하여, Dao에서 상세 스키마를 모르게 만들었습니다. Spring 공식 문서에서 찾아보니 스프링 부트가 schema.sql과 data.sql 파일을 로드시켜 sql 스크립트를 실행해준다는 것을 알게되었네요!!

Laterality commented on src/main/java/chess/domain/ChessGameService.java:
```diff
@@ -0,0 +1,151 @@
+import org.springframework.stereotype.Service;
+
+@Service
+public class ChessGameService {
```
이 클래스에 대한 테스트도 작성해보면 어떨까요?

Hongdonggeon replied:
Dao 가짜 객체를 이용해서 테스트 코드 작성해보았습니다!!

Laterality reviewed (changes requested):
안녕하세요 쿤,

리뷰어 에단입니다 :)

아쉽게도(?), 1단계는 프레임워크를 전환해보는 것이 주된 목표인 만큼 코멘트를 많이 남겨드리진 않았어요.

미션의 요구사항이 기존의 프레임워크를 '대체'하는 것이니, 기존의 웹 프레임워크와 DAO는 제거해주세요.

남겨드린 코멘트 확인해주시고, 궁금하신 점은 언제든 DM 주세요.

Hongdonggeon committed:
feat : schema.sql 적용하면서, 각 Dao init 로직에서 테이블 생성하는 sql 삭제

Hongdonggeon committed:
fix: 종료되고 재시작 되지 않는 에러 수정

reset에서 position, piece 들을 추가해주었기 때문에, 시작버튼을 누를때 또 다시 추가되어 중복 primary key 로 인한 에러발생

Hongdonggeon committed:
test: Service 테스트 코드 추가

결과 출력문 white, black 순서 바뀌는 것 수정
hashmap -> linkedHashmap 사용

Hongdonggeon committed:
remove : 불필요한 파일 삭제 및 파일 이동

Hongdonggeon commented:
콘솔, spark 웹, spring 부트 모두로 실행할 수 있게 만드는 것으로 착각했네요!!
남겨주신 리뷰 반영하였고, 현재 spring boot 에 필요하지 않은 파일들 삭제하였습니다!!
감사합니다 에단 😃

Laterality commented on src/main/java/chess/dao/JdbcTemplateBoardDao.java:
```diff
@@ -0,0 +1,51 @@
+
+    @Override
+    public void init(Map<String, String> board) {
+        String sql = "insert into board (position, piece) values (?,?)";
```
쿼리를 상수로 분리하면 어떨까요?

Laterality commented on src/main/java/chess/dto/ScoreDto.java:
```diff
@@ -0,0 +1,47 @@
+            if (value.getKey().isNone()) {
+                continue;
+            }
+            sb.append(value.getKey() + COLON + value.getValue() + SCORE_GUIDE_MESSAGE);
```
표현 방식을 정하는 것은 DTO의 역할에 부합할까요?

Hongdonggeon replied:
표현 방식을 정하는 것은 View의 역할이고 Dto는 View에 필요한 데이터만 담고 있어야 될 것 같네요!! 리뷰 반영했습니다!

Laterality reviewed (approved):
안녕하세요 쿤.

잘 수정해주셨어요 👍 

이번 미션은 이만 머지하도록 할게요.

간단한 코멘트 남겨드렸어요. 다음 단계 진행하면서 참고해주세요.

고생 많으셨습니다. 다음 단계도 화이팅하세요 :)
